File: app/src/main/java/com/veganbeauty/app/features/community/blog/BlogAdapter.java
package com.veganbeauty.app.features.community.blog;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.veganbeauty.app.R;

import java.util.List;

public class BlogAdapter extends RecyclerView.Adapter<BlogAdapter.BlogViewHolder> {

    public interface OnItemClickListener {
        void onItemClick(BlogPost post);
    }

    private List<BlogPost> posts;
    private final OnItemClickListener onItemClick;

    public BlogAdapter(List<BlogPost> posts, OnItemClickListener onItemClick) {
        this.posts = posts;
        this.onItemClick = onItemClick;
    }

    public static class BlogViewHolder extends RecyclerView.ViewHolder {
        public final ImageView ivPostImage;
        public final TextView tvPostCategory;
        public final TextView tvPostTitle;
        public final TextView tvPostDesc;

        public BlogViewHolder(View view) {
            super(view);
            ivPostImage = view.findViewById(R.id.ivPostImage);
            tvPostCategory = view.findViewById(R.id.tvPostCategory);
            tvPostTitle = view.findViewById(R.id.tvPostTitle);
            tvPostDesc = view.findViewById(R.id.tvPostDesc);
        }
    }

    @NonNull
    @Override
    public BlogViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.com_item_blog_post, parent, false);
        return new BlogViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull BlogViewHolder holder, int position) {
        BlogPost post = posts.get(position);
        holder.tvPostTitle.setText(post.getTitle());
        
        String category = (post.getCategory() != null && !post.getCategory().isEmpty()) ? post.getCategory() : "Dưỡng da";
        holder.tvPostCategory.setText(category);
        
        holder.tvPostDesc.setText("🕒 " + post.getDate() + " " + post.getDescription());
        
        if (post.getImageUrl() != null && !post.getImageUrl().isEmpty()) {
            com.bumptech.glide.Glide.with(holder.ivPostImage.getContext()).load(post.getImageUrl()).error(R.color.gray_light).into(holder.ivPostImage);
        } else {
            holder.ivPostImage.setImageResource(R.color.gray_light);
        }
        
        holder.itemView.setOnClickListener(v -> onItemClick.onItemClick(post));
    }

    @Override
    public int getItemCount() {
        return posts.size();
    }

    public void updateData(List<BlogPost> newPosts) {
        this.posts = newPosts;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/veganbeauty/app/features/community/blog/BlogAdapter.kt
package com.veganbeauty.app.features.community.blog

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import coil.load
import coil.transform.RoundedCornersTransformation
import com.veganbeauty.app.R

class BlogAdapter(
    private var posts: List<BlogPost>,
    private val onItemClick: (BlogPost) -> Unit
) : RecyclerView.Adapter<BlogAdapter.BlogViewHolder>() {

    class BlogViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val ivPostImage: ImageView = view.findViewById(R.id.ivPostImage)
        val tvPostCategory: TextView = view.findViewById(R.id.tvPostCategory)
        val tvPostTitle: TextView = view.findViewById(R.id.tvPostTitle)
        val tvPostDesc: TextView = view.findViewById(R.id.tvPostDesc)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BlogViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.com_item_blog_post, parent, false)
        return BlogViewHolder(view)
    }

    override fun onBindViewHolder(holder: BlogViewHolder, position: Int) {
        val post = posts[position]
        holder.tvPostTitle.text = post.title
        holder.tvPostCategory.text = post.category.ifEmpty { "Dưỡng da" }
        holder.tvPostDesc.text = "🕒 ${post.date} ${post.description}"
        if (post.imageUrl.isNotEmpty()) {
            holder.ivPostImage.load(post.imageUrl) {
                crossfade(true)
                error(R.color.gray_light)
                placeholder(R.color.gray_light)
            }
        } else {
            holder.ivPostImage.setImageResource(R.color.gray_light)
        }
        
        holder.itemView.setOnClickListener {
            onItemClick(post)
        }
    }

    override fun getItemCount() = posts.size

    fun updateData(newPosts: List<BlogPost>) {
        posts = newPosts
        notifyDataSetChanged()
    }
}
